azuread/table_azuread_directory_role.go

```diff
@@ -0,0 +1,133 @@
+
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+)
+
+//// TABLE DEFINITION
+
+func tableAzureAdDirectoryRole() *plugin.Table {
+	return &plugin.Table{
+		Name:        "azuread_directory_role",
+		Description: "Represents an Azure Active Directory (Azure AD) directory role",
+		Get: &plugin.GetConfig{
+			Hydrate:           getAdDirectoryRole,
+			KeyColumns:        plugin.SingleColumn("id"),
+			ShouldIgnoreError: isNotFoundError,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAdDirectoryRoles,
+			KeyColumns: plugin.KeyColumnSlice{
+				// Key fields
+				{Name: "id", Require: plugin.Optional},
+			},
+		},
+
+		Columns: []*plugin.Column{
+			{Name: "id", Type: proto.ColumnType_STRING, Description: "The unique identifier for the directory role.", Transform: transform.FromGo()},
+			{Name: "description", Type: proto.ColumnType_STRING, Description: "The description for the directory role."},
+			{Name: "display_name", Type: proto.ColumnType_STRING, Description: "The display name for the directory role."},
+			{Name: "filter", Type: proto.ColumnType_STRING, Transform: transform.FromQual("filter"), Description: "Odata query to search for groups."},
+
+			// Other fields
+			{Name: "role_template_id", Type: proto.ColumnType_STRING, Description: "The id of the directoryRoleTemplate that this role is based on. The property must be specified when activating a directory role in a tenant with a POST operation. After the directory role has been activated, the property is read only."},
+
+			// // Json fields
+			{Name: "member_ids", Type: proto.ColumnType_JSON, Hydrate: getAdDirectoryRoleMembers, Transform: transform.FromValue(), Description: "Id of the owners of the application. The owners are a set of non-admin users who are allowed to modify this object."},
+			// // Standard columns
+			// {Name: "tags", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTags, Transform: transform.From(applicationTags)},
+			{Name: "title", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTitle, Transform: transform.FromField("DisplayName", "ID")},
+			{Name: "tenant_id", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTenant, Hydrate: plugin.HydrateFunc(getTenantId).WithCache(), Transform: transform.FromValue()},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listAdDirectoryRoles(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	session, err := GetNewSession(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	client := msgraph.NewDirectoryRolesClient(session.TenantID)
+	client.BaseClient.Authorizer = session.Authorizer
+
+	directoryRoles, _, err := client.List(ctx)
+	if err != nil {
+		if isNotFoundError(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	for _, directoryRoles := range *directoryRoles {
+		d.StreamListItem(ctx, directoryRoles)
+	}
+
+	return nil, err
+}
+
+// Hydrate Functions
+
+// we didn't add the get function as it retries 5 times on 404 errors
```

Comment:
What get function are we missing, since it seems like we have the basic get function below?